tests/unit/valgrind.suppressions

```diff
@@ -10,6 +12,50 @@
    fun:pthread_create@@GLIBC_2.2.5
    fun:main
 }
+{
+   pthread_false_positive_glibc_2_34
+   Memcheck:Leak
+   match-leak-kinds: possible
+   fun:calloc
+   fun:calloc
+   fun:allocate_dtv
+   fun:_dl_allocate_tls
+   fun:allocate_stack
+   fun:pthread_create@@GLIBC_2.34
+   ...
+   fun:main
+}
```

Comment:
The reason I put two suppression blocks for different GLIBC versions is because the previous Valgrind test is ran on Ubuntu18 which will be deprecated. This CTest memcheck will run on Ubutu22 and 24 (once the Dockerfile is ready). The new suppression block is targeted specifically for GLIBC versions in Ubuntu 22 and 24. So I wrote two blocks and plan to remove the first one, once Ubutun 18 is fully deprecated.

I will keep this comment in mind and talk with James to decide if we are going to commit this suggestion.

Reply:
This change will make the code easily to maintain in the future. I will update this part in the next commit.